crates/forge/src/lib.rs

```diff
@@ -435,14 +408,67 @@ fn run_tests_from_crate(
     ))
 }
 
-fn run_with_fuzzing(
-    package_root: &Utf8PathBuf,
-    runner_config: &RunnerConfig,
-    runner_params: &RunnerParams,
-    runner: &SierraCasmRunner,
-    case: &TestCase,
-    args: &Vec<&ConcreteTypeId>,
-) -> Result<(TestCaseSummary, u32)> {
+fn choose_test_strategy_and_run(
+    case: Arc<TestCase>,
+    runner: Arc<SierraCasmRunner>,
+    runner_config: Arc<RunnerConfig>,
+    args: Vec<ConcreteTypeId>,
+) -> JoinHandle<Result<TestCaseSummary>> {
+    if args.is_empty() {
+        run_single_test(case, runner, runner_config)
+    } else {
+        tokio::task::spawn(async move {
+            run_with_fuzzing(runner_config.clone(), runner, case, &args).await
+        })
+    }
+}
```

Comment:
This is inconsistent. We should either be spawning a new task in both if and else arms or in none.

Reply:
This suggests that `run_single_test` is not actually running within a task

Reply:
I could move this `tokio::task::spawn` in to run_with_fuzzing but this function has 130lines and is very messy